### ISSUE TITLE
Weapons not saving in crates/vehicals correctly

### ISSUE THREAD

**Issue by AlexH0661** [A3Wasteland/ArmA3_Wasteland.Tanoa#34]
This seems to of happened after the latest update. I'm running the latest SQL DB (1.06).

If a player stores weapons in the hummingbird they disappear after 2-3 mins.
RPGs if loaded with a round, lose that round when stored in a crate.

I think you touched on the issue with things being stored with attachments in a previous issue ticket, and said that its to do with the actual game and how bohemia have implemented storage. Is this the same with ammunition?

Cheers,
Huss

**Comment by AgentRev** [comment]
I don't know about heli weapons, it's not an A3W component.

The RPG issue is the same as attachments. While the attachments and loaded magazines could be saved (they are not at the moment), they cannot be restored directly on weapons, because there is no scripting command to assemble custom weapons. I could restore the individual components, however I don't do that because some weapon classnames have linked attachments. For example, the gunstore GM6 Lynx has a linked LRPS, and if I restore the individual components then it would create an LRPS-equipped GM6 Lynx plus an extra LRPS, and it would create add another extra LRPS on every server restart, until the crate is full.

This problem however does not affect magazines since all bare weapons are unloaded, so in theory I could have it re-add a separate rocket alongside the empty launcher. The problem is I would have to completely rewrite the inventory saving procedure.

**Comment by AlexH0661** [comment]
No problems, I thought it might be something along those lines .

Cheers :)

**Comment by AgentRev** [comment]
I was bored so I decided to write a fix: https://github.com/A3Wasteland/ArmA3_Wasteland.Tanoa/commit/25cd65ffc60b1dfefb9f50c8035cfb747498dcb7

To test, I placed an accessorized pistol in my backpack, and put the backpack in storage, alongside a loaded RPG. This is how it got restored:

![image](https://user-images.githubusercontent.com/1380241/34284792-a0fc1c38-e6a2-11e7-94ac-d27936557057.png)

So it breaks down everything into separate items, which occupy the same total inventory capacity as the original combined items. Pretty much the closest I can get. Also applies to vehicles and crates.
